### PR TITLE
Adding endpoint type option to the extension

### PR DESCRIPTION
**Pull request by rakeshvanga**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: src/dev-spaces/azext_dev_spaces/custom.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # pylint:disable=no-member,too-many-lines,too-many-locals,too-many-statements
 
 
-def ads_use_dev_spaces(cluster_name, resource_group_name, update=False, space_name=None, do_not_prompt=False):
+def ads_use_dev_spaces(cluster_name, resource_group_name, update=False, space_name=None, endpoint_type=None, do_not_prompt=False):
```
Comment (amsoedal) [inline src/dev-spaces/azext_dev_spaces/custom.py]: endpoint_type=Public

**Comment by rakeshvanga** [comment]
This PR is ready to be checked in. @zikalino can you merge these changes?

**Comment by rakeshvanga** [comment]
> You may need to re-build and re-upload wheel to you private contianer.
> image

Updated the latest whl file to our container. @haroldrandom , can you review and merge it now?

**Comment by haroldrandom** [comment]
Sure, but why added label “do not merge”?

**Comment by rakeshvanga** [comment]
@yonzhan added this label due to miscommunication. There is no need to wait on this PR.

File: src/dev-spaces/azext_dev_spaces/azext_metadata.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-    "azext.minCliCoreVersion": "2.0.63"
```
Comment (haroldrandom) [inline src/dev-spaces/azext_dev_spaces/azext_metadata.json]: If you set `"azext.minCliCoreVersion": "2.1.0"`, the extension will be only available when azure-cli >= 2.1.0.
Are you sure about it?.

Comment (rakeshvanga) [reply src/dev-spaces/azext_dev_spaces/azext_metadata.json]: Yes. I have validated the changes with the 2.1.0 version.

**Pull request merged**